compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -468,6 +460,52 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
         })
     }
+
+    fn check_offset_align(
+        &self,
+        offset: u64,
+        align: Align,
+        check: CheckAlignment,
+    ) -> InterpResult<'tcx> {
+        if offset % align.bytes() == 0 {
+            Ok(())
+        } else {
+            // The biggest power of two through which `offset` is divisible.
+            let offset_pow2 = 1 << offset.trailing_zeros();
+            self.check_alignment(Align::from_bytes(offset_pow2).unwrap(), align, check)
+        }
+    }
+
+    fn check_alignment(
```

Comment:
This function throws an alignment violation error -- please rename it to reflect that. Maybe `alignment_check_failed` or so?